src/Service.Tests/Configuration/LoadConfigViaEndpoint.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Service.Controllers;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static Azure.DataApiBuilder.Service.Tests.Configuration.ConfigurationEndpoints;
+using static Azure.DataApiBuilder.Service.Tests.Configuration.TestConfigFileReader;
+
+namespace Azure.DataApiBuilder.Service.Tests.Configuration;
+
+[TestClass]
+public class LoadConfigViaEndpointTests
+{
+    [TestMethod("Testing that missing environment variables won't cause runtime failure."), TestCategory(TestCategory.COSMOSDBNOSQL)]
+    [DataRow(CONFIGURATION_ENDPOINT_V2)]
+    public async Task CanLoadConfigWithMissingEnvironmentVariables(string configurationEndpoint)
+    {
+        TestServer server = new(Program.CreateWebHostFromInMemoryUpdateableConfBuilder(Array.Empty<string>()));
+        HttpClient httpClient = server.CreateClient();
+
+        RuntimeConfig config = ReadCosmosConfigurationFromFile() with { Schema = "@env('schema')" };
+
+        ConfigurationPostParametersV2 @params = new(
+            Configuration: config.ToJson(),
+            ConfigurationOverrides: "{}",
+            Schema: @"
+            type Entity {
+                id: ID!
+                name: String!
+            }
+            ",
+            AccessToken: null
+        );
+
+        JsonContent content = JsonContent.Create(@params);
+
+        HttpResponseMessage postResult =
+            await httpClient.PostAsync(configurationEndpoint, content);
+        Assert.AreEqual(HttpStatusCode.OK, postResult.StatusCode);
+
+        RuntimeConfigProvider configProvider = server.Services.GetService(typeof(RuntimeConfigProvider)) as RuntimeConfigProvider;
+        RuntimeConfig loadedConfig = configProvider.GetConfig();
+
+        Assert.AreEqual(config.Schema, loadedConfig.Schema);
+    }
+
+    [TestMethod("Testing that environment variables can be replaced at runtime not only when config is loaded."), TestCategory(TestCategory.COSMOSDBNOSQL)]
+    [DataRow(CONFIGURATION_ENDPOINT_V2)]
+    public async Task CanLoadConfigWithEnvironmentVariables(string configurationEndpoint)
```

Comment:
Now that I think of it: do we really want to allow replacement of the environment variables at all if they exist when the config is sent to the endpoint? Before 0.8.49, we would attempt to replace the environment variables only when the source of config is the local file system. 

A side effect would be any default environment variable already set on the container will now be available for consumption - e.g. `@env('ASPNETCORE_ENVIRONMENT')`

Reply:
If we don't want to do env replacement then it's a much simpler fix, as we can disable that code path entirely for the HTTP config setting. I assumed that SWA would have some things they send in, such as the connection info.